test/spec/modules/kargoBidAdapter_spec.js

```diff
@@ -1,461 +1,1481 @@
+      localStorage.setItem(name, value);
+    }
+
+    const crbValues = {
+      valid: buildCrbValue(true, true, true, true, true, false),
+      validLs: buildCrbValue(false, true, true, true, true, false),
+      invalidB64: 'crbValue',
+      invalidB64Ls: 'crbValueLs',
+      invalidJson: 'eyJ0ZXN0OiJ2YWx1ZSJ9',
+      invalidJsonLs: 'eyJ0ZXN0OiJ2YWx1ZSJ9',
+    };
+    function setCrb(
+      setCookie = 'valid',
+      setLocalStorage = 'valid'
+    ) {
+      if (crbValues.hasOwnProperty(setCookie)) {
+        setCookieValue('krg_crb', crbValues[setCookie]);
+      }
+      if (crbValues.hasOwnProperty(`${setLocalStorage}Ls`)) {
+        setLocalStorageValue('krg_crb', crbValues[`${setLocalStorage}Ls`]);
+      }
+    }
+
+    it('exists and produces an object', function() {
+      const request = spec.buildRequests(bids, bidderRequest);
+      expect(request).to.exist.and.to.be.an('object');
+    });
+
+    it('produces a POST request with a payload', function() {
+      const request = spec.buildRequests(bids, bidderRequest);
+      expect(request.method).to.exist.and.equal('POST');
+      expect(request.url).to.exist.and.equal('https://krk2.kargo.com/api/v1/prebid');
+
+      const payload = request.data;
+      expect(payload).to.exist;
+      expect(payload.imp).to.have.length(2);
+
+      // Display bid
+      const bidImp = payload.imp[0];
+      expect(bidImp.id).to.equal('randomBidId');
+      expect(bidImp.banner).to.deep.equal({ sizes: [ [970, 250], [1, 1] ] });
+      expect(bidImp.video).to.be.undefined;
+      expect(bidImp.bidRequestCount).to.equal(1);
+      expect(bidImp.bidderRequestCount).to.equal(1);
+      expect(bidImp.code).to.equal('displayAdunitCode');
+      expect(bidImp.ext.ortb2Imp).to.deep.equal(defaultBidParams.ortb2Imp);
+      expect(bidImp.fpd).to.deep.equal({ gpid: '/1234/prebid/slot/path' });
+      expect(bidImp.pid).to.equal('displayPlacement');
+
+      // Video bid
+      const videoBidImp = payload.imp[1];
+      expect(videoBidImp.id).to.equal('randomBidId2');
+      expect(videoBidImp.banner).to.be.undefined;
+      expect(videoBidImp.video).to.deep.equal(outstreamBid.mediaTypes.video);
+      expect(videoBidImp.bidRequestCount).to.equal(1);
+      expect(videoBidImp.bidderRequestCount).to.equal(1);
+      expect(videoBidImp.code).to.equal('instreamAdunitCode');
+      expect(videoBidImp.ext.ortb2Imp).to.deep.equal(defaultBidParams.ortb2Imp);
+      expect(videoBidImp.fpd).to.deep.equal({ gpid: '/1234/prebid/slot/path' });
+      expect(videoBidImp.pid).to.equal('instreamPlacement');
+
+      // User
+      expect(payload.user).to.be.an('object');
+      expect(payload.user.crbIDs).to.deep.equal({});
+      expect(payload.user.data).to.deep.equal([]);
+      expect(payload.user.sharedIDEids).to.deep.equal(defaultBidParams.userIdAsEids);
+
+      // General keys
+      expect(payload.aid).to.equal('randomAuctionId');
+      expect(payload.device).to.deep.equal({ size: [ 3440, 1440 ] });
```

Comment:
This one is failing on my end, definitely because the adapter is referencing `window` to get this information. Maybe  let's just set a device h/w using that `getPayloadFromTestBids` function you set up.
https://github.com/KargoGlobal/Prebid.js/blob/master/modules/kargoBidAdapter.js#L91-L94

Reply:
I opted to just pull the values from the window object rather than set them on the window object to avoid mutating the global or having to set up a new sandbox to override the window global without affecting other adapter tests, thoughts on that approach?